libs/pubsub/pubsub_test.go

```diff
@@ -402,12 +415,10 @@ func TestBufferCapacity(t *testing.T) {
 	require.ErrorIs(t, s.Publish(ctx, "Ironclad"), context.DeadlineExceeded)
 }
 
-func newTestServer(ctx context.Context, t testing.TB) *pubsub.Server {
+func newTestServer(ctx context.Context, logger log.Logger, t testing.TB) *pubsub.Server {
```

Comment:
Nit: I think ctx and t should precede other parameters.
```suggestion
func newTestServer(ctx context.Context, t testing.TB, logger log.Logger) *pubsub.Server {
```